tests/api2/test_initshutdownscript.py

```diff
@@ -0,0 +1,81 @@
+import base64
+import errno
+import os
+import stat
+import uuid
+
+import pytest
+
+from auto_config import ip, ha
+from middlewared.test.integration.utils.client import client
+from middlewared.service_exception import ValidationErrors, ValidationError
+
+TEST_SCRIPT_FILE = '/tmp/.TEST_SCRIPT_FILE'
+_775 = stat.S_IRWXU | stat.S_IRWXG | stat.S_IROTH | stat.S_IXOTH
+
+
+@pytest.fixture(scope='module')
+def ip_to_use():
+    return ip if not ha else os.environ['controller1_ip']
+
+
+@pytest.fixture(scope='module')
+def ws_client(ip_to_use):
+    with client(host_ip=ip_to_use) as c:
+        yield c
+
+
+def test_initshtudown_script(ws_client):
+    # create the test script file first
+    ws_client.call(
+        'filesystem.file_receive',
+        TEST_SCRIPT_FILE,
+        base64.b64encode(b'echo "testing"').decode(),
```

Comment:
You can just use `ssh` command to create simple files on the NAS

Reply:
That's true but I'm trying to stick to only using websocket client for this module